addons/sale/static/src/js/product_catalog/kanban_record.js

```diff
@@ -48,11 +48,6 @@ export class ProductCatalogKanbanRecord extends KanbanRecord {
             product_id: this.env.productId,
             quantity: this.state.quantity,
         })
-        // Force a reload of the page to remove a product from the view in the case a filter is
-        // applied.
-        if (this.state.quantity === 0) {
-            this.props.record.model.load();
-        }
```

Comment:
This was causing the concurrence issue when removing a product from the view and was costly performance wise, removing it can only be positive.